### PR TITLE
chalabi/modal fixes

### LINKED ISSUE #28: Modal closable with Esc
All modals should be closable by pressing the `Esc` key.

### LINKED ISSUE #31: Link in the transaction detail should point to the block explorer
The correct block explorer should be used and depends on the selected endpoint.

Using a local instance, the link is incorrectly pointing to 

http://localhost:3000/undefined/transaction/D55D7A5DEF7326B673E7096BABABE51D95D73EF46109B5F14868612AD2D0D54A

### LINKED ISSUE #64: Improve real estate usage when adding a new Group member
We should be able to handle more than 2 members without having to scroll. A group could contain dozens of members, if not more.

### PR DESCRIPTION
**Pull request by chalabi2**
partially fixes #28 (pending token factory modal logic rework)
fixes #64
fixes #31

## Release Notes

- **New Features**
	- Enhanced readability of large transaction amounts with a new formatting function.
	- Streamlined modal handling for transaction and denomination information.
  
- **Bug Fixes**
	- Improved error handling in modals with optional chaining for safer property access.
  
- **User Interface Improvements**
	- Adjusted maximum height for member list display in the Member Management Modal.
  
- **Refactor**
	- Simplified modal props and internal logic across multiple components for better maintainability.

### CODE REVIEW

File: components/bank/components/tokenList.tsx [hunk 1]
```diff
@@ -95,7 +92,10 @@
                     <button
                       onClick={e => {
                         e.stopPropagation();
-                        openModal(balance.metadata);
+                        setSelectedDenom(balance?.denom);
+                        (
+                          document?.getElementById(`denom-info-modal`) as HTMLDialogElement
+                        )?.showModal();
```
[the remark this answers is not in the document]
Comment (chalabi2) [reply components/bank/components/tokenList.tsx:98]: this causes the user to have to click twice

File: components/bank/components/historyBox.tsx [hunk 2]
```diff
@@ -184,7 +197,12 @@
                       <div
                         key={tx.tx_hash}
                         className="flex items-center justify-between p-4 bg-[#FFFFFFCC] dark:bg-[#FFFFFF0F] rounded-[16px] cursor-pointer hover:bg-[#FFFFFF66] dark:hover:bg-[#FFFFFF1A] transition-colors"
-                        onClick={() => openModal(tx)}
+                        onClick={() => {
+                          setSelectedTx(tx);
+                          (
+                            document?.getElementById(`tx_modal_info`) as HTMLDialogElement
+                          )?.showModal();
+                        }}
```
[the remark this answers is not in the document]
Comment (chalabi2) [reply components/bank/components/historyBox.tsx:205]: this is reccomended usage in daisyui

File: components/bank/components/historyBox.tsx [hunk 3]
```diff
@@ -254,7 +273,7 @@
         </div>
       )}
 
-      {selectedTx && <TxInfoModal tx={selectedTx} isOpen={!!selectedTx} onClose={closeModal} />}
+      <TxInfoModal modalId={`tx_modal_info`} tx={selectedTx ?? ({} as TransactionGroup)} />
```
[the remark this answers is not in the document]
Comment (chalabi2) [reply components/bank/components/historyBox.tsx:276]: there wont be undefined tx info as the only time a tx object gets rendered is if there is tx data for it.

File: components/groups/modals/memberManagementModal.tsx [hunk 4]
```diff
@@ -223,7 +223,7 @@ export function MemberManagementModal({
                       <div className="w-[45%]">Address</div>
                       <div className="w-[10%]"></div>
                     </div>
-                    <div className="space-y-4 max-h-[420px] overflow-y-auto">
+                    <div className="space-y-4 max-h-[22rem] overflow-y-auto">
```
[the remark this answers is not in the document]
Comment (chalabi2) [reply components/groups/modals/memberManagementModal.tsx:226]: this is fixed in other pr

**Review by fmorency — APPROVED** [review]

**Pull request merged**